react-hash-routing/client/app.jsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react';
+import Header from './components/header';
+import Catalog from './pages/catalog';
+import NotFound from './pages/not-found';
+import ProductDetails from './pages/product-details';
+import { parseRoute } from './lib';
+
+export default function App() {
+  const [route, setRoute] = useState(parseRoute(window.location.hash));
+
+  useEffect(() => {
+    window.addEventListener('hashchange', () => {
+      setRoute(parseRoute(window.location.hash));
+    });
+    return () => window.removeEventListener('hashchange', setRoute(parseRoute(window.location.hash)));
```

Comment:
You can't remove an event listener like this. You have to pass the exact same function to both `add` and `remove`. So declare the function first, then pass it to both.